include/CircularDependencyDetector.h

```diff
@@ -27,7 +28,7 @@ namespace ChimeraTK::detail {
     void unregisterDependencyWait(VariableNetworkNode& node);
 
     /// Print modules which are currently waiting for initial values.
-    void printWaiters();
+    void printWaiters(std::ostream& stream = std::cout);
```

Comment:
it's a bit confusing that here, you choose std::cout as default, instead of the ChimeraTK::Logger. I guess the reason is that the function is going to be used as debug output about threads behavior, rather than application output? Maybe a comment is in place.

Reply:
I cannot use the logger, since we have to take a reference to the `std::ostream` here (it is not copyable) but the logger returns a new object. I will add a comment.